clients/src/main/java/org/apache/kafka/clients/consumer/AcknowledgeType.java

```diff
@@ -25,4 +25,17 @@ public enum AcknowledgeType {
     public String toString() {
         return super.toString().toLowerCase(Locale.ROOT);
     }
+
+    public static AcknowledgeType acknowledgeTypeFromNumber(int a) {
+        switch (a) {
```

Comment:
Hmmm, might be better to associate with an id in constructor and change method name to `forId` as that's being don in other enums like CompressionType.java.